java/piranha/src/main/java/com/uber/piranha/XPFlagCleaner.java

```diff
@@ -368,31 +367,43 @@ private API getXPAPI(ExpressionTree et, VisitorState state) {
     return API.UNKNOWN;
   }
 
-  private Matcher<MethodInvocationTree> methodRecordMatcher(
-      MethodRecord methodRecord, VisitorState state) {
+  private boolean methodRecordMatcher(
+      MethodRecord methodRecord, VisitorState state, MethodInvocationTree mit) {
 
-    return allOf(
-        (methodRecord.isStatic() ? staticMethod().anyClass() : instanceMethod().anyClass())
-            .named(methodRecord.getMethodName()),
-        methodRecord
-            .getArgumentIdx()
-            .map(idx -> argument(idx, isArgumentMatchesFlagName(state)))
-            .orElseGet(Matchers::anything),
-        methodRecord
+    return // Method name must match methodRecord.getMethodName()
```

Comment:
If we are returning a boolean anyways, can't we make this a fair bit clearer by using conditionals vs a 20+ lines boolean expression? E.g.

```
// Method name must match methodRecord.getMethodName()
Matcher<MethodInvocationTree> nameMatcher = 
           (methodRecord.isStatic() ? staticMethod().anyClass() : instanceMethod().anyClass())
            .named(methodRecord.getMethodName());
if (!nameMatcher.matches(mit, state)) {
   return false;
}
```

and so on?

Reply:
Somehow GitHub swallowed my follow up comment here, but one point I want to make is about debugability: it's way easier to step through the code or set a breakpoint if this entire method is not a single statement.

Reply:
I agree. This will ease debugging through code too. I ll actually split it.